python-package/xgboost/core.py

```diff
@@ -806,7 +823,7 @@ def save_binary(self, fname, silent=True) -> None:
                                              c_str(fname),
                                              ctypes.c_int(silent)))
 
-    def set_label(self, label) -> None:
+    def set_label(self, label: Collection) -> None:
```

Comment:
Not sure why is this a `Collection`?  Maybe ArrayLike?

Reply:
My bad; I was confused by `dispatch_meta_backend` but apparently I missed the docstring...